user_management/src/user/views.py

```diff
@@ -0,0 +1,165 @@
+                                    password=json_request['password'])
+                refresh_token = JWTManager('refresh').generate_token(user.id)
+                return JsonResponse(data={'refresh_token': refresh_token}, status=201)
+            else:
+                return JsonResponse(data={'errors': validation_errors}, status=400)
+        except json.JSONDecodeError:
+            return JsonResponse(data={'errors': ['Invalid JSON format in the request body']}, status=400)
+
+        except Exception as e:
+            return JsonResponse(data={'errors': ['An unexpected error occurred']}, status=500)
+
+    def signup_infos_validation(self, json_request):
+        validation_errors = []
+
+        username = json_request.get('username')
+        email = json_request.get('email')
+        password = json_request.get('password')
+
+        valid_username, error_message_username = self.is_valid_username(username)
+        valid_email, error_message_email = self.is_valid_email(email)
+        valid_password, error_message_password = self.is_valid_password(password)
+
+        if not valid_username:
+            validation_errors.append(error_message_username)
+        if not valid_email:
+            validation_errors.append(error_message_email)
+        if not valid_password:
+            validation_errors.append(error_message_password)
+        return validation_errors
+
+    @staticmethod
+    def is_valid_username(username):
+        if username is None or username == '':
+            return False, 'Username empty'
+        if len(username) < settings.USERNAME_MIN_LENGTH:
+            return False, f'Username length {len(username)} < {settings.USERNAME_MIN_LENGTH}'
+        if len(username) > settings.USERNAME_MAX_LENGTH:
+            return False, f'Username length {len(username)} > {settings.USERNAME_MAX_LENGTH}'
+        users = User.objects.filter(username=username)
+        if users.exists():
+            return False, f'Username {username} already taken'
+        return True, None
+
+    @staticmethod
+    def is_valid_email(email):
+        if email is None or email == '':
+            return False, 'Email empty'
+        if len(email) > settings.EMAIL_MAX_LENGTH:
+            return False, f'Email length {len(email)} > {settings.EMAIL_MAX_LENGTH}'
+        if any(char in '!#$%^&*()=' for char in email):
+            return False, f'Invalid character in email address'
+        if '@' not in email:
+            return False, 'Email missing @'
+        if '.' not in email:
+            return False, 'Email missing "." character'
+        if email.count('@') > 1:
+            return False, 'Email contains more than one @ character'
+        local_part, _ = email.rsplit('@', 1)
+        if len(local_part) < settings.EMAIL_LOCAL_PART_MIN_LENGTH:
+            return False, f'Local part length {len(local_part)} < {settings.EMAIL_LOCAL_PART_MIN_LENGTH}'
+        _, domain_and_tld = email.rsplit('@', 1)
```

Comment:
I'm approving it so you don't have to wait for me, but change this so you don't split the same string twice for no good reason